FakeDatabase/campaign-data.json

```diff
@@ -0,0 +1,296 @@
+{
+	"campaigns": [
+		{
+			"name": "Finding Screamo",
+			"key": "a12d3",
+			"gameMaster": "Tom",
+			"players": [
```

Comment:
If we want to practice flattening and keying, in prep for the real DB, then I'm surprised to see us nesting a few things.

Consider a separate list for players, sessions, and currentGoals.

For currentGoals in particular, consider just goals. It might be useful to have a status on them.  Active vs completed vs abandoned.  I imagine the completed goals being crossed out would be a satisfying thing to see in the UI.

Reply:
I'm used to using no-SQL databases in personal projects, so I don't worry about making table-shaped objects.  If we're planning on SQL, I can restructure.  I do think the shape of the data for the project is consistent enough to lend itself well to SQL.

For quest goals: I don't see historical goals as being relevant.  Goals are too flexible and subject to change.  Too often, the player or GM will start with a path in mind and the story just doesn't go that way.  Unlike a programmed video game, a TTRPG mutates unexpectedly by the wills of the players and GM.

Consider this scenario: the GM has a plot about a quest giver actually being the big bad.  The GM plans for the quest giver to send the players to three areas of increasing difficulty until the quest giver does a big reveal as the boss of the third area.

planned goals:
- find area 1
- explore area 1
- defeat the area 1 boss

In the actual game, the players reach the first boss.  However, instead of engaging in combat, they start talking to the boss.  The boss ends up accusing the quest giver of being the boss's slave owner.  The players are torn.  Some are ready to believe the boss.  Some believe the boss is trying to trick them.  The players make the boss promise not to do bad things while they return to town to investigate the matter further.

Defeat the area 1 boss hasn't necessarily been abandoned: the players may still do that.  But it's no longer the current goal.  The current goal is to investigate the quest giver.

In essence, current goals are in-the-moment intensions.  Ones that can change dramatically and rapidly for many reasons.  They're not the same as a task list of concrete steps you want to check off.

Reply:
Thanks for the excellent explanation.  

SQL vs no-SQL is something we can decide later.  For now, this data shape looks great for consumption by the front end.